plugins/outputs/riemann/riemann.go

```diff
@@ -68,8 +79,16 @@ func (r *Riemann) Write(metrics []telegraf.Metric) error {
 
 	var senderr = r.client.SendMulti(events)
 	if senderr != nil {
```

Comment:
in here I would say just do:

```go
r.Close()
return fmt.Errorf("FAILED to send riemann message (will try to reconnect). Error: %s\n",
				senderr)
```

because r.Close will set the connection to nil, and then a connection will be attempted on the next write.